snuba/query/parser/__init__.py

```diff
@@ -103,3 +128,75 @@ def _parse_query_impl(body: MutableMapping[str, Any], dataset: Dataset) -> Query
         having=having_expr,
         order_by=orderby_exprs,
     )
+
+
+def _validate_aliases(query: Query) -> None:
+    """
+    Ensures that no alias has been defined multiple times for different
+    expressions in the query. Thus rejecting queries with shadowing.
```

Comment:
We are still allowing duplicate aliases as long as they reference the same expression right? Should we have a test to clarify this behavior is supported?

Reply:
> We are still allowing duplicate aliases as long as they reference the same expression right?

Correct

> Should we have a test to clarify this behavior is supported?

Good idea I will add one to the test_query. If it is not yet there, I believe it may already be in the test case.

Reply:
done